src/java.desktop/share/classes/javax/swing/JTable.java

```diff
@@ -2146,42 +2146,51 @@ public boolean getCellSelectionEnabled() {
         return getRowSelectionAllowed() && getColumnSelectionAllowed();
     }
 
-    /**
-     *  Selects all rows, columns, and cells in the table.
-     */
-    public void selectAll() {
-        // If I'm currently editing, then I should stop editing
-        if (isEditing()) {
-            removeEditor();
-        }
-        if (getRowCount() > 0 && getColumnCount() > 0) {
-            int oldLead;
-            int oldAnchor;
-            ListSelectionModel selModel;
+    private void selectRows(ListSelectionModel selModel, int rowCount) {
+        selModel.setValueIsAdjusting(true);
+        int oldLead = getAdjustedIndex(selModel.getLeadSelectionIndex(), true);
+        int oldAnchor = getAdjustedIndex(selModel.getAnchorSelectionIndex(), true);
 
-            selModel = selectionModel;
-            selModel.setValueIsAdjusting(true);
-            oldLead = getAdjustedIndex(selModel.getLeadSelectionIndex(), true);
-            oldAnchor = getAdjustedIndex(selModel.getAnchorSelectionIndex(), true);
+        setRowSelectionInterval(0, rowCount - 1);
 
-            setRowSelectionInterval(0, getRowCount()-1);
+        // this is done to restore the anchor and lead
+        SwingUtilities2.setLeadAnchorWithoutSelection(selModel, oldLead, oldAnchor);
+
+        selModel.setValueIsAdjusting(false);
+    }
 
-            // this is done to restore the anchor and lead
-            SwingUtilities2.setLeadAnchorWithoutSelection(selModel, oldLead, oldAnchor);
+    private void selectColumns(TableColumnModel  columnModel, int columnCount) {
```

Comment:
Should we check whether column selection is allowed before selecting columns? Because by default only row selection is enabled.

Reply:
No, as told earlier, selectAll selects all cells...
WHere do you see only row selection is enabled? In SwingSet2 JTable, if you do ctrl+A which is equivalent to selectAll, all cells gets selected..

Reply:
[Here](https://github.com/openjdk/jdk/blob/577ede73d8e916bac9050d3bee80d2f18cc833a7/src/java.desktop/share/classes/javax/swing/JTable.java#L5629)

Reply:
ok..But selectAll selects all cells as per the spec Spec says "Selects all rows, columns, and cells in the table..

Reply:
In SwingSet2 JTable demo before the fix also, you can see only "rowSelection" is enabled, but selectAll selects all cells in the table so selectAll overrides that setting..

Reply:
Ok.

Reply:
Added columnSelection check..SwingSet2 selectAll selection can be viewed as selecting all rows which can be perceived as selecting all cells, so to be safe added the check..

Reply:
> Added columnSelection check..SwingSet2 selectAll selection can be viewed as selecting all rows which can be perceived as selecting all cells, so to be safe added the check..

Moreover, It is seen that by unchecking "rowSelection" check selectAll didnt allow cell selection so it doesnt override the row/column selection-allowed setting..so added the columnSelectionAllowed check before selecting columns and similarly for rows..